test/new-e2e/utils/infra/stack_manager.go

```diff
@@ -61,53 +64,61 @@ func newStackManager(ctx context.Context) (*StackManager, error) {
 }
 
 // GetStack creates or return a stack based on env+stack name
-func (sm *StackManager) GetStack(ctx context.Context, envName string, name string, config auto.ConfigMap, deployFunc pulumi.RunFunc, failOnMissing bool) (*auto.Stack, auto.UpResult, error) {
+func (sm *StackManager) GetStack(ctx context.Context, name string, config runner.ConfigMap, deployFunc pulumi.RunFunc, failOnMissing bool) (*auto.Stack, auto.UpResult, error) {
 	sm.lock.RLock()
 	defer sm.lock.RUnlock()
 
-	var getStack func(context.Context, string, string, pulumi.RunFunc, ...auto.LocalWorkspaceOption) (auto.Stack, error)
-	if failOnMissing {
-		getStack = auto.SelectStackInlineSource
-	} else {
-		getStack = auto.UpsertStackInlineSource
-	}
+	// Build configuration from profile
+	profile := runner.GetProfile()
```

Comment:
I don't see a clear benefit outside of testability, which is something we currently don't do because of other pieces (need to mock Pulumi Automation API)